espnet2/uasr/segmenter/abs_segmenter.py

```diff
@@ -0,0 +1,21 @@
+from abc import ABC, abstractmethod
+
```

Comment:
Can you add some descriptions somewhere (task?) what `segmenter` is?

Reply:
Sure.

Practically, the output of the generator (in frame-level) may predict the same phoneme for consecutive frames, which makes it too easy for the discriminator. So, the segmenter here is to merge frames with a similar prediction from the generator output. The segmenter could have many variants depending on how to merge frames (e.g., use the mean, do sampling, etc.). For now, we support the one with the best performance in the wav2vec-u paper.